### ISSUE TITLE
Browser can't handle lots of js files

### ISSUE THREAD

**Issue by mjarkk** [btd/rollup-plugin-visualizer#103]
First of all i love the library and it has saved helpt give a lot of usefull inside!

Now my problem, i have been splitting up code a lot lately and the more chunks i add the slower this stats page became till the point where i can't even open it anymore :(.

## Rollup config

So i don't have a rollup config as i'm using vite so here is my vite config:
```js
import { createVuePlugin } from "vite-plugin-vue2";
import { visualizer } from 'rollup-plugin-visualizer';

export default ({ command }) => {
    const production = command !== 'serve'

    const plugins = [
        createVuePlugin({
            vueTemplateOptions: {
                transformAssetUrls: false,
            },
        }),
    ];

    if (production) plugins.push(visualizer())

    return {
        build: {
            assetsInclude: [],
            manifest: true,
            outDir: 'public/js',
            rollupOptions: {
                input: 'resources/js/main.js'
            }
        },
        plugins,
        server: {
            port: 3000,
        },

        publicDir: 'resources',
    };
};
```

## Node and dependencies versions

- node: v15.5.0
- rollup-plugin-visualizer: 5.4.1
- vite: 2.2.1 *(imported rollup 2.45.2)*

## Your OS type and version
MacOS 11.2.3

## Screenshot of problem

![image](https://user-images.githubusercontent.com/15320763/116468176-d4819e00-a870-11eb-8e2b-503907745e5e.png)

## How my build output looks
As this is a private code base i sadly cannot share everything so here is a "vagueified" build output:

<details>
<summary>Output:</summary>
<br>

```
public/js/assets/_______.1e5109e8.js    4.59kb / brotli: 1.78kb
public/js/assets/_______.5f1d9086.js    42.50kb / brotli: 12.29kb
public/js/assets/_______.5d38324a.js    5.28kb / brotli: 1.77kb
public/js/assets/_______.f8a2ff5d.js    0.71kb / brotli: 0.36kb
public/js/assets/_______.2b7fa1d9.js    2.59kb / brotli: 0.91kb
public/js/assets/_______.2bad36ac.js    0.51kb / brotli: 0.29kb
public/js/assets/_______.4a2115d5.js    5.75kb / brotli: 1.83kb
public/js/assets/_______.85aab3cc.js    1.00kb / brotli: 0.41kb
public/js/assets/_______.42904f14.js    0.14kb / brotli: 0.11kb
public/js/assets/_______.59e8ddea.js    0.34kb / brotli: 0.17kb
public/js/assets/_______.c64bef2a.js    1.30kb / brotli: 0.60kb
public/js/assets/_______.a935b3ee.js    42.14kb / brotli: 11.62kb
public/js/assets/_______.5deb0905.js    1.58kb / brotli: 0.66kb
public/js/assets/_______.a32245f8.js    7.60kb / brotli: 1.85kb
public/js/assets/_______.f8ded0d2.js    1.88kb / brotli: 0.73kb
public/js/assets/_______.af4d88a6.js    0.90kb / brotli: 0.41kb
public/js/assets/_______.40323d90.js    8.06kb / brotli: 2.40kb
public/js/assets/_______.be546988.js    1.16kb / brotli: 0.52kb
public/js/assets/_______.2ff8c1ce.js    5.13kb / brotli: 1.04kb
public/js/assets/_______.f76de28f.js    7.35kb / brotli: 1.73kb
public/js/assets/_______.f17f735a.js    0.13kb / brotli: 0.11kb
public/js/assets/_______.3e1ea335.js    17.49kb / brotli: 4.44kb
public/js/assets/_______.aa9202b8.js    2.43kb / brotli: 0.94kb
public/js/assets/_______.1c8e5206.js    1.02kb / brotli: 0.44kb
public/js/assets/_______.44857de0.js    3.33kb / brotli: 1.11kb
public/js/assets/_______.1ba3cf2a.js    1.54kb / brotli: 0.69kb
public/js/assets/_______.269f2ae7.js    12.94kb / brotli: 3.90kb
public/js/assets/_______.6bec32d1.js    1.34kb / brotli: 0.61kb
public/js/assets/_______.7313eaff.js    3.59kb / brotli: 1.40kb
public/js/assets/_______.9e33a77f.js    1.84kb / brotli: 0.81kb
public/js/assets/_______.acfabebc.js    15.61kb / brotli: 3.32kb
public/js/assets/_______.3a5fd8f5.js    13.45kb / brotli: 3.27kb
public/js/assets/_______.de4bd521.js    2.27kb / brotli: 0.97kb
public/js/assets/_______.1ab4530a.js    0.93kb / brotli: 0.46kb
public/js/assets/_______.c1d77f76.js    0.57kb / brotli: 0.24kb
public/js/assets/_______.8f4c6d1c.js    43.57kb / brotli: 12.31kb
public/js/assets/_______.0717551e.js    1.27kb / brotli: 0.59kb
public/js/assets/_______.ffbb2e87.js    1.40kb / brotli: 0.66kb
public/js/assets/_______.e299c31e.js    8.51kb / brotli: 2.46kb
public/js/assets/_______.ca6c5b9a.js    3.49kb / brotli: 1.04kb
public/js/assets/_______.572e635c.js    7.37kb / brotli: 2.08kb
public/js/assets/_______.8de63b2f.js    2.29kb / brotli: 0.93kb
public/js/assets/_______.eb07bd5c.js    1.79kb / brotli: 0.63kb
public/js/assets/_______.6900e95c.js    0.33kb / brotli: 0.22kb
public/js/assets/_______.017dcf30.js    2.28kb / brotli: 0.89kb
public/js/assets/_______.9d8e2226.js    0.06kb / brotli: 0.07kb
public/js/assets/_______.f90dbca5.js    1.36kb / brotli: 0.61kb
public/js/assets/_______.5b8c7d1c.js    2.74kb / brotli: 1.22kb
public/js/assets/_______.c21c7a8c.js    7.63kb / brotli: 2.23kb
public/js/assets/_______.c98963f8.js    1.29kb / brotli: 0.59kb
public/js/assets/_______.2ba90ca8.js    1.30kb / brotli: 0.60kb
public/js/assets/_______.be897773.js    7.59kb / brotli: 2.28kb
public/js/assets/_______.38b50f32.js    1.55kb / brotli: 0.71kb
public/js/assets/_______.e98b43f6.js    15.79kb / brotli: 3.89kb
public/js/assets/_______.3c248319.js    19.51kb / brotli: 5.33kb
public/js/assets/_______.30dc9eec.js    1.21kb / brotli: 0.58kb
public/js/assets/_______.021516ae.js    11.45kb / brotli: 2.87kb
public/js/assets/_______.ed65917a.js    9.59kb / brotli: 2.82kb
public/js/assets/_______.468c383d.js    36.49kb / brotli: 6.91kb
public/js/assets/_______.4f3876a3.js    8.74kb / brotli: 2.35kb
public/js/assets/_______.ecb38807.js    2.30kb / brotli: 0.96kb
public/js/assets/_______.4425e9e3.js    12.10kb / brotli: 2.56kb
public/js/assets/_______.7c40cfd1.js    11.31kb / brotli: 2.67kb
public/js/assets/_______.2f8eb6eb.js    9.62kb / brotli: 2.91kb
public/js/assets/_______.880fc265.js    9.36kb / brotli: 3.94kb
public/js/assets/_______.c13b645d.js    0.41kb / brotli: 0.19kb
public/js/assets/_______.24a0df70.js    0.54kb / brotli: 0.20kb
public/js/assets/_______.ab150007.js    2.61kb / brotli: 1.08kb
public/js/assets/_______.45af380d.js    0.08kb / brotli: 0.06kb
public/js/assets/_______.27add46e.js    0.36kb / brotli: 0.14kb
public/js/assets/_______.3b0f0d5b.js    0.05kb / brotli: 0.05kb
public/js/assets/_______.f6749717.js    0.28kb / brotli: 0.14kb
public/js/assets/_______.8047b4e7.js    0.77kb / brotli: 0.18kb
public/js/assets/_______.54fb107d.js    0.73kb / brotli: 0.21kb
public/js/assets/_______.00fc3dab.js    2.79kb / brotli: 0.68kb
public/js/assets/_______.62d6ccf1.js    0.65kb / brotli: 0.19kb
public/js/assets/_______.20c9f85c.js    0.63kb / brotli: 0.23kb
public/js/assets/_______.d40abeec.js    2.40kb / brotli: 0.50kb
public/js/assets/_______.aba55d52.js    15.40kb / brotli: 4.62kb
public/js/assets/_______.dc16a33b.js    85.83kb / brotli: 20.92kb
public/js/assets/_______.85dabe57.js    373.79kb / brotli: 72.96kb
public/js/assets/_______.74dba5e8.css   455.88kb / brotli: 38.61kb
public/js/assets/_______.5c5b1c3b.js    413.16kb / brotli: 68.82kb
public/js/assets/_______.f49591bb.js    469.59kb / brotli: 99.77kb
public/js/assets/_______.b54cb711.js    627.77kb / brotli: skipped (large chunk)
```

</details>

## Other notes:

The stats file it generated was 15,9MB

**Comment by mjarkk** [comment]
Could there maybe be an option to only include files in the stats file that are larger than x KB because when i use this tool i mainly care about the biggest files?

**Comment by mjarkk** [comment]
Found for now a temporary solution by using the sunburst template.
Loading is almost instant tough i'd still be rather using treemap.

**Comment by btd** [comment]
hi, i thought sooner or later i will get such an issue.
So you have 85 bundles. How many files approximately you have ? You can say, by opening stats file (or just generating json).
there at root of json should be nodes object. you can get the number of keys.
Because now it is not very clear in which part of code perfomance problem. I want first to reproduce the problem and next decide if i should add options. Options is a last resort in my mind.

**Comment by mjarkk** [comment]
31007 nodes are in the stats file,  
Now that i'm looking at the json stats file i think i can replace all names and send it to you.

**Comment by btd** [comment]
> Now that i'm looking at the json stats file i think i can replace all names and send it to you.

**Comment by mjarkk** [comment]
I've renamed all path parts with hashes and after a long time waiting and a few retries the browser finally showed the stats.
Can't add json files to github comments so i placed the file in a repo:
https://github.com/mjarkk/rollup-plugin-visualizer-stats

**Comment by mjarkk** [comment]
Maybe it's worth adding a "veagufier" to this package when installing it globally,

<details>
<summary>I used this to "veagufiy" the stats file:</summary>
<br>

```js
const fs = require('fs')
const crypto = require('crypto')

function hash(value) {
    const shasum = crypto.createHash('sha1')
    shasum.update(value)
    return shasum.digest('hex').slice(0, 15)
}

function hashPath(value) {
    return value.split('/').map(part => !part ? part : part == 'node_modules' ? part : hash(part)).join('/')
}

const file = fs.readFileSync('./stats.json', { encoding: 'utf8' })
let conf = JSON.parse(file)

//
// NODES
//

for (const [key, value] of Object.entries(conf.nodes)) {
    conf.nodes[key].id = hashPath(value.id);
}

//
// NODE PARTS
//

for (const [key, value] of Object.entries(conf.nodeParts)) {
    let newMap = {}
    Object.entries(value).map(([file, id]) => {
        newMap[hashPath(file)] = id;
    })
    conf.nodeParts[key] = newMap
}

//
// TREE
//

function parseChildren(children) {
    return children.map(child => {
        child.name = hashPath(child.name)
        if (child.children) {
            child.children = parseChildren(child.children)
        }
        return child;
    })
}
conf.tree.children = parseChildren(conf.tree.children)

fs.writeFileSync('./stats.vague.json', JSON.stringify(conf, null, 2))
```

</details>

**Comment by btd** [comment]
Thank you. I will take a look. 
Nice idea about veagufiy. Maybe it is worth to include something like this to this plugin, this will make debugging issues much more simple.

**Comment by btd** [comment]
I just published 5.5.0, can you please give it a try? If it is still slow can you pls post json, generated with this plugin version.

**Comment by mjarkk** [comment]
Holyshit it opens instantly now.
Thanks!!
